core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,321 @@
+        );
+
+        assert_eq!(
+            base.try_subtract(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(base),
+                operator: NumericBinaryOperator::Subtract,
+                rhs: Bool(true).clone(),
+            }
+            .into()),
+        );
+    }
+
+    #[test]
+    fn try_multiply() {
+        let base = Decimal::ONE;
+
+        assert!(matches!(base.try_multiply(&I8(1)), Ok(Decimal(x)) if x == Decimal::ONE));
+        assert!(matches!(base.try_multiply(&I64(1)), Ok(Decimal(x)) if x == Decimal::ONE));
+        assert!(matches!(base.try_multiply(&F64(1.0)), Ok(Decimal(x)) if x == Decimal::ONE));
+        assert!(
+            matches!(base.try_multiply(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ONE)
+        );
+
+        assert_eq!(
+            base.try_multiply(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(base),
+                operator: NumericBinaryOperator::Multiply,
+                rhs: Bool(true).clone(),
+            }
+            .into()),
+        );
+    }
+
+    #[test]
+    fn try_divide() {
+        let base = Decimal::ONE;
+
+        assert!(
+            matches!(base.try_divide(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ONE)
+        );
+        assert!(
+            matches!(base.try_divide(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ONE)
+        );
+        assert!(
+            matches!(base.try_divide(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ONE)
+        );
+        assert!(
+            matches!(base.try_divide(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ONE)
+        );
+
+        assert_eq!(
+            base.try_divide(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(base),
+                operator: NumericBinaryOperator::Divide,
+                rhs: Bool(true).clone(),
+            }
+            .into()),
+        );
+    }
+
+    #[test]
+    fn try_modulo() {
+        let base = Decimal::ONE;
+
+        assert!(
+            matches!(base.try_modulo(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ZERO)
+        );
+        assert!(
+            matches!(base.try_modulo(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ZERO)
+        );
+        assert!(
+            matches!(base.try_modulo(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ZERO)
+        );
+        assert!(
+            matches!(base.try_modulo(&Decimal(Decimal::ONE)), Ok(Decimal(x)) if x == Decimal::ZERO)
+        );
+
+        assert_eq!(
+            base.try_modulo(&Bool(true)),
+            Err(ValueError::NonNumericMathOperation {
+                lhs: Decimal(base),
+                operator: NumericBinaryOperator::Modulo,
+                rhs: Bool(true).clone(),
```

Comment:
```suggestion
                rhs: Bool(true),
```

Reply:
Usually `clone` is used to clone a variable, but it doesn't seem necessary here.

Another story) I wonder why this is not caught in clippy error.

Reply:
+1